homeassistant/components/sensor/__init__.py

```diff
@@ -476,6 +477,23 @@ def state(self) -> Any:
                     f"Sensor {self.entity_id} provides state value '{value}', "
                     "which is not in the list of options provided"
                 )
+            return value
+
+        # If the sensor has neither a device class, a state class nor
+        # a unit_of measurement then there are no further checks or conversions
+        if not device_class and not state_class and not unit_of_measurement:
```

Comment:
I think the check must ensure the device class is a known device class in the `SensorDeviceClass` enum as well.

Custom integration will use custom device classes for legacy style translations. Additionally, the end-user customize feature allows for just strings as device classes as well.

We we can only validate what we know.